### PR TITLE
Feature/components demo

### PR DESCRIPTION
**Pull request by mis4mike**
This PR is the first example of styling components in line with [issue #93](https://github.com/BSA-US/dual-power-app/issues/93). It adds a checkbox and button-bar-style radio buttons. 

If you will be visually styling components per the Manhattan Hydraulics mock-ups, this is a good example. Check out the changes in:
- components/radioButtons.tsx - Shows how to apply a `treatment` prop to a UI component to give it a particular treatment
- styles/radio-buttons.js - Shows how to write WindiCSS shortcuts to create general shared styles. Styles specific to a particular component should be added as classes to that component's `className` prop (See checkbox.tsx - notice that the CSS module is sparse, providing only styling not currently supported by WindiCSS).
- windi.config.ts - Adds the shortcut so that it can be applied globally

Other changes:
- Add a component demo page
- Adds props to existing UI components
- Adds a checkbox component
- Adds support for importing SVGs for next/image

### CODE REVIEW

File: components/checkbox.tsx [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+  CheckboxProps,
+  CheckboxIndicatorProps,
+} from '@radix-ui/react-checkbox'
 
-export const Checkbox = (props: any) => (
-  <Root {...props} className={`${props.className}`}></Root>
+import styles from './checkbox.module.css'
+
+export const Checkbox = ({ className = '', ...props }: CheckboxProps) => (
+  <Root {...props} className={`${styles.checkbox} ${className}`}></Root>
```
Comment (replygirl) [inline components/checkbox.tsx:10]: point of windicss is to use utility classes directly in the classname

File: components/radioButtons.tsx [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+  RadioGroupProps,
+  RadioGroupItemProps,
+  RadioIndicatorProps,
+} from '@radix-ui/react-radio-group'
 
-export const RadioButtons = (props: any) => (
-  <Root {...props} className={`${props.className}`} />
-)
+export interface RadioButtonsProps extends RadioGroupProps {
+  variant: 'radio-button-bar' | '' | undefined
```
Comment (replygirl) [inline components/radioButtons.tsx:9]: dont pass in a shortcut—shortcuts are utility classes that can be applied to other components. if there are other formats planned for this component i suggest one of two things:
- change the first literal in this type to `’bar’`
- use separate components, calling this one RadioButtonsBar

Comment (replygirl) [reply components/radioButtons.tsx:9]: is there a different word besides variant that wouldn’t conflict with terminology in windicss? maybe layout or format

File: components/radioButtons.tsx [hunk 3]
```diff
@@ -1,12 +1,28 @@
 import { Root, Item, Indicator } from '@radix-ui/react-radio-group'
+import type {
```
Comment (replygirl) [inline components/radioButtons.tsx:2]: not in scope for this pr necessarily but filenames should all be param-case

File: package.json [hunk 4]
```diff
@@ -32,6 +32,7 @@
     "faunadb": "^4.3.0",
     "isomorphic-unfetch": "^3.1.0",
     "next": "^11.0.0",
+    "next-react-svg": "^1.1.3",
```
Comment (replygirl) [inline package.json:35]: theres a conflict here but i think rebasing onto main will be an easy one here

File: package.json [hunk 5]
```diff
@@ -16,10 +16,10 @@
   "dependencies": {
     "@notionhq/client": "^0.2.1",
     "@peertube/embed-api": "^0.0.4",
-    "@radix-ui/react-avatar": "^0.0.15",
-    "@radix-ui/react-checkbox": "^0.0.15",
-    "@radix-ui/react-radio-group": "^0.0.15",
-    "@radix-ui/react-tabs": "^0.0.15",
+    "@radix-ui/react-avatar": "^0.1.0",
```
Comment (replygirl) [inline package.json:19]: they hit minor! good news :)

File: styles/radioButtons.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const radioButtonBar = {
```
Comment (replygirl) [inline styles/radioButtons.js:1]: filenames should be param-case

Comment (replygirl) [reply styles/radioButtons.js:1]: if this shortcut were kept (it should be removed, see other comments) it should either be kept directly in the windicss config, or, when we have more than a couple dozen shortcuts, more ALL of the shortcuts into one file as a default or single const export

File: components/checkbox.module.css [hunk 7]
```diff
@@ -0,0 +1,22 @@
+.checkbox {
+  position: relative;
+  height: 24px;
+  width: 24px;
+  border: 1px solid #d1d6dc;
+  border-radius: 4px;
+
+  &:after {
```
Comment (replygirl) [inline components/checkbox.module.css:8]: windicss has an `:after` variant for targeting this https://windicss.org/utilities/variants.html#pseudo-elements

File: components/checkbox.module.css [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    content: "";
+    position: absolute;
+    top: -5px;
+    left: -5px;
+    height: calc(100% + 10px);
+    width: calc(100% + 10px);
+    border: 2px solid #dc2626;
+    border-radius: 8px;
+  }
+  &[aria-checked="true"] {
```
Comment (replygirl) [inline components/checkbox.module.css:18]: windi doesn’t have a variant for this nor is there a plugin, so we should keep this ruleset here for now, but since i did a little looking into it:

it turns out their plugin api [makes it pretty easy](https://windicss.org/plugins/interfaces.html#add-variants) to add a variant.

i also put in an issue proposing that aria attributes could be supported in the core variants so e.g. we could use `checked:*` here: https://github.com/windicss/windicss/issues/555

File: components/checkbox.module.css [hunk 9]
```diff
@@ -0,0 +1,22 @@
+.checkbox {
```
Comment (replygirl) [inline components/checkbox.module.css:1]: this should all be in the classname

File: components/checkbox.module.css [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  position: relative;
+  height: 24px;
+  width: 24px;
+  border: 1px solid #d1d6dc;
```
Comment (replygirl) [inline components/checkbox.module.css:5]: if this isnt in the windi theme maybe u should just go for it and add it to the colors?

File: components/checkbox.module.css [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    position: absolute;
+    top: -5px;
+    left: -5px;
+    height: calc(100% + 10px);
+    width: calc(100% + 10px);
+    border: 2px solid #dc2626;
+    border-radius: 8px;
+  }
+  &[aria-checked="true"] {
+    background: url(/images/checkmark.svg) no-repeat center;
```
Comment (replygirl) [inline components/checkbox.module.css:19]: every rule in this file except the `background-image` on this line and the `calc`s can be added with `@apply windi-class-foo bar-baz` etc.

and if windi had a support for aria-checked we’d be able to do `checked:background-image-[url(/images/checkmark.svg/)]`

related, i would like to add iconify svg with auto-import but im also kind of holding out for windicss to just add them directly, as v4 will be adopting most of [unocss’s new goodies](https://antfu.me/posts/icons-in-pure-css) (anthony fu, who made this, is also on the windi team)

File: components/checkbox.module.css [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+  width: 24px;
+  border: 1px solid #d1d6dc;
+  border-radius: 4px;
+
+  &:after {
+    content: "";
+    position: absolute;
+    top: -5px;
+    left: -5px;
+    height: calc(100% + 10px);
```
Comment (replygirl) [inline components/checkbox.module.css:13]: `calc`s might work in `className` as `’h-[calc(100% + 10px)] w-[calc(100% + 10px)]’—i dont actually remember on that one

why isnt this possible with negative margin though? `’after:(content absolute inset-0 -m-5)’` should give you the same box

File: windi.config.ts [hunk 13]
```diff
@@ -56,6 +60,9 @@ export default defineConfig({
       ),
     extend: {
       screens: {
```
Comment (replygirl) [inline windi.config.ts:62]: these screens are already provided by windicss—extend should prevent our theme from overwriting them, was that not the case?

**Review by replygirl — CHANGES_REQUESTED** [review]

File: components/radioButtons.tsx (hunk 2 above)
Comment (mis4mike) [reply components/radioButtons.tsx]: Changed to "treatment" and changed the title to "button-bar", which could see use other places.

Comment (mis4mike) [reply components/radioButtons.tsx]: Changed the prop name to "treatment"

**Comment by jackhajb** [comment]
@mis4mike On focus I'm noticing the components all get a blue outline.  Think we should override.

image

File: components/checkbox.tsx [hunk 14]
```diff
@@ -1,9 +1,21 @@
...
+} from '@radix-ui/react-checkbox'
 
-export const Checkbox = (props: any) => (
-  <Root {...props} className={`${props.className}`}></Root>
+import styles from './checkbox.module.css'
+
+export const Checkbox = ({ className = '', ...props }: CheckboxProps) => (
+  <Root
+    {...props}
+    className={`relative w-24px h-24px border-1 border-cool-gray-300 rounded-sm after:(content absolute inset-0 -m-5px rounded-md border-2 border-red-600) ${styles.checkbox} ${className}`}
```
Comment (jackhajb) [inline components/checkbox.tsx:12]: Can we multi-line this?

Comment (mis4mike) [reply components/checkbox.tsx:12]: For some reason I'm unable to reply to the comment above about the focus outline. I am down to override the focus indicator styling [as long as it's cross-browser and accessible](https://coryrylan.com/blog/dont-override-css-outline-focus-styles). By all means feel free to make this part of a component ticket 😸

Comment (replygirl) [reply components/checkbox.tsx]: @jackhajb feel free to multiline and organize classes on your own branches. eslint will respect your decisioins

i'm not going to do it on this branch, but i work in vue SFCs a lot where my eslint configs automatically add line breaks and group related classes, so i see where you're coming from

another thing to consider, that i haven't used much yet or made sure is working in our repos, is [attributify mode](https://windicss.org/posts/attributify.html). assuming its working, you could also switch over from `className` to attributes once you have enough styles to cause a line break, or need a more manageable way to bind

tl;dr theres wiggle room here

**Comment by replygirl** [comment]
turns out windicss plugins are super easy to make:

- [`aria-*:foo` variants (`aria-checked`, `not-aria-checked`, `aria-selected, `not-aria-selected`)](https://github.com/BSA-US/dual-power-app/blob/7c6ed4a10222a5dfc2ce56141dd25fbfef978a8c/packages/%40replygirl/windicss-plugins/aria.ts)
- [`content:foo` variant](https://github.com/BSA-US/dual-power-app/blob/7c6ed4a10222a5dfc2ce56141dd25fbfef978a8c/packages/%40replygirl/windicss-plugins/content.ts)

File: components/checkbox.tsx (hunk 14 above)
Comment (jackhajb) [reply components/checkbox.tsx]: @mis4mike I hear you on accessibility, let's hold off for now.  A custom focus outline would probably be best, but not really a priority at this point

Comment (jackhajb) [reply components/checkbox.tsx]: @replygirl Attributify mode seems interesting.  Would you rather have a line length of 200 char or 15 lines of attributes. Not sure my thoughts on that one.  The line break was a soft suggestion, so Mike will add or not as he sees fit.

Comment (mis4mike) [reply components/checkbox.tsx]: We probably do want to set a maximum line length. I don't have strong opinions yet - I think we can pick something that seems reasonable and see how it works with our code.

**Comment by mis4mike** [comment]
@replygirl does this PR reflect all your changes, and if so, is it ready to merge? The code and Vercel environment look good to me, but Github seems to think there's a pending change request somewhere that I can't find.

**Comment by replygirl** [comment]
> @mis4mike On focus I'm noticing the components all get a blue outline. Think we should override.
> 
> image

I added a `focus-ring` shortcut class that will handle this for us :)

File: components/checkbox.tsx (hunk 14 above)
Comment (replygirl) [reply components/checkbox.tsx]: i don't have a stake in it since we have 80ch globally enforced everywhere except long strings. going to mark as resolved and consider out of scope for this pr. if either of u want to extend the eslint config to give whitespace in jsx special treatment, feel free to do so in a different pr

**Review by replygirl — APPROVED** [review]

**Comment by replygirl** [comment]
I approve of my own changes lol
